TX.h

```diff
@@ -396,85 +574,45 @@ void setup(void)
 #else
   Serial.begin(115200);
 #endif
-  setupProfile();
-  txReadEeprom();
-
-  setupPPMinput();
-  ppmAge = 255;
 
   setupRfmInterrupt();
 
   sei();
 
-  start = millis();
-  while ((ppmAge == 255) && ((millis() - start) < 2000));
+  setupProfile();
+  txReadEeprom();
 
   buzzerOn(BZ_FREQ);
   digitalWrite(BTN, HIGH);
   Red_LED_ON ;
 
-  while (Serial.available()) {
-    Serial.read();
-  }
+  consoleFlush();
 
-  Serial.print("OpenLRSng TX starting ");
-  printVersion(version);
-  Serial.print(" on HW ");
-  Serial.println(BOARD_TYPE);
+  consolePrint("OpenLRSng TX starting ");
+  printVersion(version, getConsoleSerial());
+  consolePrint(" on HW BOARD_TYPE\n");
 
-  delay(50);
+  delay(100);
 
   checkBND();
 
-  if (bind_data.serial_baudrate && (bind_data.serial_baudrate < 5)) {
-    serialMode = bind_data.serial_baudrate;
-    if(serialMode == 3) {
-      //SBUS, 100Kbaud, 8E2
-      TelemetrySerial.begin(100000, SERIAL_8E2);
-    } else {
-      TelemetrySerial.begin(115200);
-    }
-  } else {
-    // switch to userdefined baudrate here
-    TelemetrySerial.begin(bind_data.serial_baudrate);
-  }
   checkButton();
 
   Red_LED_OFF;
   buzzerOff();
 
-  setupPPMinput(); // need to do this to make sure ppm polarity is correct if profile was changed
+  setupRcSerial();
 
-  altPwrIndex=0;
-  if(tx_config.flags & ALT_POWER) {
-    if (bind_data.hopchannel[6] && bind_data.hopchannel[13] && bind_data.hopchannel[20]) {
-      altPwrIndex=7;
-    } else {
-      altPwrIndex=5;
-    }
+  uint32_t timeout = millis() + (serialMode == SERIAL_MODE_MULTI ? 20000 : 2000);
+  while (ppmAge == 255 && millis() < timeout) {
```

Comment:
When this wait loop starts, PPM hasn't been inited yet so it will always wait 2 seconds. setupPPMinput() should probably be returned to a place below setupRcSerial() above. The condition serialMode == SERIAL_MODE_NONE can applied.